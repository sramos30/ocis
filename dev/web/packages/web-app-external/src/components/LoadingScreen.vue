<template>
  <div class="oc-text-center">
    <oc-spinner size="xlarge" />
    <p v-translate class="oc-invisible">Loading app</p>
  </div>
</template>

<script>
export default {
  name: 'LoadingScreen'
}
</script>
