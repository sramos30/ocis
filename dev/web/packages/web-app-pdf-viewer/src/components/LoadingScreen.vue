<template>
  <div class="oc-position-center oc-text-center">
    <oc-spinner size="xlarge" />
    <p v-translate class="oc-invisible">Loading PDF document</p>
  </div>
</template>

<script>
export default {
  name: 'LoadingScreen'
}
</script>
