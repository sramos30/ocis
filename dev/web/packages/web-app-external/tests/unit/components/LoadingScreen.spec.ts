import LoadingScreen from '../../../src/components/LoadingScreen.vue'
import { Wrapper, mount, createLocalVue } from '@vue/test-utils'
import GetTextPlugin from 'vue-gettext'

const localVue = createLocalVue()

localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})

let wrapper: Wrapper<any>

describe('The external app loading screen component', () => {
  test('displays a spinner and a paragraph', () => {
    wrapper = mount(LoadingScreen, {
      localVue,
      stubs: {
        OcSpinner: true
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
