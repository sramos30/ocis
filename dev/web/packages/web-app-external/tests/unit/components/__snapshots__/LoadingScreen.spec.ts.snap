// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The external app loading screen component displays a spinner and a paragraph 1`] = `
<div class="oc-text-center">
  <ocspinner-stub size="xlarge"></ocspinner-stub>
  <p class="oc-invisible" data-msgid="Loading app" data-current-language="en_US">Loading app</p>
</div>
`;
